R/pkg/R/sparkR.R

```diff
@@ -365,6 +365,23 @@ sparkR.session <- function(
     }
     overrideEnvs(sparkConfigMap, paramMap)
   }
+  if (!grepl(".*shell\\.R$", Sys.getenv("R_PROFILE_USER"), perl = TRUE)) {
```

Comment:
I think what @felixcheung meant was that in the Java side we already have a check called `isLocal`[1]. Thus we can just call that function using `callJMethod` instead of writing a new one

[1] https://github.com/apache/spark/blob/338a98d65c8efe0c41f39a8dddeab7040dcda125/core/src/main/scala/org/apache/spark/SparkContext.scala#L244